packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -264,6 +264,19 @@ class Plugin {
 
   static List<String> _validateLegacyYaml(YamlMap yaml) {
     final List<String> errors = <String>[];
+
```

Comment:
I feel like it doesn't make sense to put these checks here as the method name says "_validateLegacyYaml", but the code is actually validating the new format.

What do you think about fixing it in the `validatePluginYaml`? We have something like:

```dart
if (usesNewPluginFormat) {
  if (yaml['platforms'] != null && yaml['platforms'] is! YamlMap) {
    const String errorMessage = 'flutter.plugin.platforms should be a map with the platform name as the key';
    return <String>[errorMessage];
  }
  return _validateMultiPlatformYaml(yaml['platforms'] as YamlMap);
} else {
  return _validateLegacyYaml(yaml);
}
```

We can make it as:

```dart
if (usesNewPluginFormat) {
  if (yaml['platforms'] != null && yaml['platforms'] is! YamlMap) {
    const String errorMessage = 'flutter.plugin.platforms should be a map with the platform name as the key';
    return <String>[errorMessage];
  }
  return _validateMultiPlatformYaml(yaml['platforms'] as YamlMap);
} else if (usesOldPluginFormat) {
  return _validateLegacyYaml(yaml);
} else {
  return <String>['The flutter.plugin.platforms key in the `pubspec.yaml` is not found. '
                              'An instruction to format the `pubspec.yaml` can be found here: ' 
                              'https://flutter.dev/docs/development/packages-and-plugins/developing-packages#plugin'];
}
```

Reply:
Yeah, put the check inside the method `_validateLegacyYaml` is weird.
I will modify it later, thanks for the suggestion : ) !

Reply:
Thanks! Feel free to @ me when it's ready to be reviewed again :)